src/lib.js

```diff
@@ -265,6 +265,103 @@ function ByteQueue(size)
 }
 
 
+/**
+ * @constructor
+ *
+ * Queue wrapper around Float32Array
+ * Used by devices such as the sound blaster sound card
+ */
+function FloatQueue(size)
+{
+    var data = new Float32Array(size),
+        start,
+        end;
+
+    dbg_assert((size & size - 1) === 0);
+
+    this.length = 0;
+
+    this.push = function(item)
+    {
+        if(this.length === size)
+        {
+            // intentional overwrite
+            start = start + 1 & size - 1;
+        }
+        else
+        {
+            this.length++;
+        }
+
+        data[end] = item;
+        end = end + 1 & size - 1;
+    };
+
+    this.shift = function()
+    {
+        if(!this.length)
+        {
+            return undefined;
+        }
+        else
+        {
+            var item = data[start];
+
+            start = start + 1 & size - 1;
+            this.length--;
+
+            return item;
+        }
+    };
+
+    this.shift_block = function(count)
+    {
+        var slice = new Float32Array(count);
+
+        if(count > this.length)
+        {
+            count = this.length;
+        }
+        var slice_end = start + count;
+
+        var partial = data.slice(start, slice_end);
```

Comment:
Don't use `.slice`, it makes an unnecessary copy. Instead, use `.subarray` (also below).